packages/aws-cdk-lib/cx-api/FEATURE_FLAGS.md

```diff
@@ -128,7 +127,8 @@ The following json shows the current recommended set of flags, as `cdk init` wou
     "@aws-cdk/aws-codepipeline:crossAccountKeysDefaultValueToFalse": true,
     "@aws-cdk/aws-codepipeline:defaultPipelineTypeToV2": true,
     "@aws-cdk/aws-kms:reduceCrossAccountRegionPolicyScope": true,
-    "@aws-cdk/aws-eks:nodegroupNameAttribute": true
+    "@aws-cdk/aws-eks:nodegroupNameAttribute": true,
+    "@aws-cdk/aws-ec2:ebsDefaultGp3Volume": true
```

Comment:
Any reason not to have something like `"@aws-cdk/aws-ec2:ebsDefaultVolume": "gp3"`? The current setup makes future changes in the default volume type a little more awkward, we'll have to have multiple `aws-cdk/aws-ec2:ebsDefaultXXXVolume` flags, make sure there are no conflicts, etc.

Reply:
Let's say if we introduce `"@aws-cdk/aws-ec2:ebsDefaultVolume": "gp3"` in this PR and after a while AWS announces `gp4` as a new default and we modify this feature flag to `"@aws-cdk/aws-ec2:ebsDefaultVolume": "gp4"`, this would cause breaking changes for those opting in `gp3` and we still need a check to see if they have explicitly enable this flag as `gp3`. I am not sure if that would be easier to maintain. Another consideration is conventionally our feature flags are boolean except for [target partitions](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/cx-api/FEATURE_FLAGS.md#aws-cdkcoretarget-partitions). Will check this with the maintainers.

Reply:
> we modify this feature flag to "@aws-cdk/aws-ec2:ebsDefaultVolume": "gp4", this would cause breaking changes for those opting in gp3 and we still need a check to see if they have explicitly enable this flag as gp3

Wouldn't we only change the recommended value to `gp4`? Users that opted in to `gp3` would be in the same situation as those who enabled `ebsDefaultGp3Volume` if we introduce `ebsDefaultGp4Volume`. Arguably even worse off, since they would both need to disable `ebsDefaultGp3Volume` and enable `ebsDefaultGp4Volume`, vs just updating `ebsDefaultVolume` to `gp4`.

I'm leaving it up to you guys, I might not be seeing the whole picture, just throwing it out there 👍

Reply:
One thing to note here is that feature flags are handled as a boolean so we can't use `gp3`.

Reply:
Oh, Pahud already commented that. I missed that on the initial read.